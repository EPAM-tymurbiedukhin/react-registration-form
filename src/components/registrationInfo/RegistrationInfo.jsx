import React from 'react';
import { setUpDatepicker, setUpTimepicker } from '../../globals/materialize-elements-setting';

import MainInput from '../mainInput';

export default ({ handleSubmit }) => {
  return (
    <form onSubmit={handleSubmit} className="form">
      <MainInput
        name="birthday"
        formName="registrationInfo"
        type="text"
        className="timepicker"
        placeholder="Input birthday"
        inputRef={setUpDatepicker}
      />
      <MainInput
        name="lunchtime"
        formName="registrationInfo"
        type="text"
        className="timepicker"
        placeholder="Choose lunchtime"
        inputRef={setUpTimepicker}
      />
      <button className="waves-effect waves-light btn">Finish</button>
    </form>
  );
};
