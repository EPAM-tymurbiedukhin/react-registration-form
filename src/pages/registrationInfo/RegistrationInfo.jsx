import React from 'react';
import { useDispatch } from 'react-redux';

import MainWrapper from '../../components/mainWrapper';
import Breadcrumbs from '../../components/navigations';
import RegistrationInfo from '../../components/registrationInfo';

import {
  setRegistrationInfoField,
  submitRegistrationInfoForm,
} from '../../features/registrationInfo/actions';

export default () => {
  const dispatch = useDispatch();

  const handleSubmit = (event) => {
    event.preventDefault();

    const inputs = event.target.querySelectorAll('input');
    inputs.forEach((input) => {
      const { name, value } = input;
      dispatch(setRegistrationInfoField({ name, value }));
    });

    dispatch(submitRegistrationInfoForm());
  };

  return (
    <MainWrapper>
      <RegistrationInfo handleSubmit={handleSubmit} />
      <Breadcrumbs />
    </MainWrapper>
  );
};
