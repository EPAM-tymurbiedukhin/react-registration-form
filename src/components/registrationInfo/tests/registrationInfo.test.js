import { mount } from 'enzyme';
import React from 'react';

import RegistrationInfo from '../RegistrationInfo';

import {
  stubRegistrationInfoProps,
  defaultValue,
  registrationInfoLinks,
} from './stubProps';

jest.mock('react-redux', () => ({
  useSelector: jest.fn(() => 'default value'), // defaultValue
}));

const setUp = (props) => mount(<RegistrationInfo {...props} />);

describe('<RegistrationInfo />', () => {
  let component;
  let form;

  beforeEach(() => {
    component = setUp(stubRegistrationInfoProps);
    form = component.find('form');
  });

  it('should form has correct class', () => {
    expect(form.hasClass('form')).toBeTruthy();
  });

  it('should call handleSubmit on submit form', () => {
    form.simulate('submit');
    expect(stubRegistrationInfoProps.handleSubmit).toHaveBeenCalled();
  });

  it('should all inputs render correctly', () => {
    component.find('input').forEach((node, i) => {
      const nodeProps = node.props();
      const { name, type, placeholder } = registrationInfoLinks[i];

      expect(nodeProps.name).toBe(name);
      expect(nodeProps.type).toBe(type);
      expect(nodeProps.id).toBe(name);
      expect(nodeProps.value).toBe(defaultValue);
      expect(nodeProps.placeholder).toBe(placeholder);
    });
  });

  it('should call submitForm on button click', () => {
    const button = component.find('button');
    button.simulate('click');
    expect(stubRegistrationInfoProps.handleSubmit).toHaveBeenCalled();
  })
});
