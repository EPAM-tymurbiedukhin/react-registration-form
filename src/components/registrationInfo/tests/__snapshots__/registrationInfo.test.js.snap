// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<RegistrationInfo /> should render correctly 1`] = `"<form class=\\"form\\"><div class=\\"input-field col s12 timepicker\\"><div class=\\"modal datepicker-modal\\" id=\\"modal-21e34c57-3664-b448-4f5b-53c20d8233f0\\" tabindex=\\"0\\"><div class=\\"modal-content datepicker-container\\"><div class=\\"datepicker-date-display\\"><span class=\\"year-text\\"></span><span class=\\"date-text\\"></span></div><div class=\\"datepicker-calendar-container\\"><div class=\\"datepicker-calendar\\"></div><div class=\\"datepicker-footer\\"><button class=\\"btn-flat datepicker-clear waves-effect\\" style=\\"visibility: hidden;\\" type=\\"button\\"></button><div class=\\"confirmation-btns\\"><button class=\\"btn-flat datepicker-cancel waves-effect\\" type=\\"button\\">Cancel</button><button class=\\"btn-flat datepicker-done waves-effect\\" type=\\"button\\">Ok</button></div></div></div></div></div><input name=\\"birthday\\" type=\\"text\\" id=\\"birthday\\" placeholder=\\"Input birthday\\" class=\\"\\" value=\\"default value\\"><label for=\\"birthday\\"></label></div><div class=\\"input-field col s12 timepicker\\"><div class=\\"modal timepicker-modal\\" id=\\"modal-d48598cc-cbde-1461-886a-99c4a4c85d8a\\" tabindex=\\"0\\"><div class=\\"modal-content timepicker-container\\"><div class=\\"timepicker-digital-display\\"><div class=\\"timepicker-text-container\\"><div class=\\"timepicker-display-column\\"><span class=\\"timepicker-span-hours text-primary\\"></span>:<span class=\\"timepicker-span-minutes\\"></span></div><div class=\\"timepicker-display-column timepicker-display-am-pm\\"><div class=\\"timepicker-span-am-pm\\"><div class=\\"am-btn\\">AM</div><div class=\\"pm-btn\\">PM</div></div></div></div></div><div class=\\"timepicker-analog-display\\"><div class=\\"timepicker-plate\\"><div class=\\"timepicker-canvas\\"><svg class=\\"timepicker-svg\\" width=\\"270\\" height=\\"270\\"><g transform=\\"translate(135,135)\\"><line x1=\\"0\\" y1=\\"0\\"></line><circle class=\\"timepicker-canvas-bg\\" r=\\"20\\"></circle><circle class=\\"timepicker-canvas-bearing\\" cx=\\"0\\" cy=\\"0\\" r=\\"4\\"></circle></g></svg></div><div class=\\"timepicker-dial timepicker-hours\\"><div class=\\"timepicker-tick\\" style=\\"left: 167.5px; top: 24.06733260263394px;\\">1</div><div class=\\"timepicker-tick\\" style=\\"left: 205.93266739736606px; top: 62.499999999999986px;\\">2</div><div class=\\"timepicker-tick\\" style=\\"left: 220px; top: 115px;\\">3</div><div class=\\"timepicker-tick\\" style=\\"left: 205.93266739736606px; top: 167.49999999999997px;\\">4</div><div class=\\"timepicker-tick\\" style=\\"left: 167.5px; top: 205.93266739736606px;\\">5</div><div class=\\"timepicker-tick\\" style=\\"left: 115px; top: 220px;\\">6</div><div class=\\"timepicker-tick\\" style=\\"left: 62.499999999999986px; top: 205.93266739736606px;\\">7</div><div class=\\"timepicker-tick\\" style=\\"left: 24.067332602633954px; top: 167.50000000000006px;\\">8</div><div class=\\"timepicker-tick\\" style=\\"left: 10px; top: 115.00000000000003px;\\">9</div><div class=\\"timepicker-tick\\" style=\\"left: 24.067332602633954px; top: 62.499999999999986px;\\">10</div><div class=\\"timepicker-tick\\" style=\\"left: 62.49999999999994px; top: 24.067332602633968px;\\">11</div><div class=\\"timepicker-tick\\" style=\\"left: 114.99999999999997px; top: 10px;\\">12</div></div><div class=\\"timepicker-dial timepicker-minutes timepicker-dial-out\\"><div class=\\"timepicker-tick\\" style=\\"left: 115px; top: 10px;\\">00</div><div class=\\"timepicker-tick\\" style=\\"left: 167.5px; top: 24.06733260263394px;\\">05</div><div class=\\"timepicker-tick\\" style=\\"left: 205.93266739736606px; top: 62.499999999999986px;\\">10</div><div class=\\"timepicker-tick\\" style=\\"left: 220px; top: 115px;\\">15</div><div class=\\"timepicker-tick\\" style=\\"left: 205.93266739736606px; top: 167.49999999999997px;\\">20</div><div class=\\"timepicker-tick\\" style=\\"left: 167.5px; top: 205.93266739736606px;\\">25</div><div class=\\"timepicker-tick\\" style=\\"left: 115px; top: 220px;\\">30</div><div class=\\"timepicker-tick\\" style=\\"left: 62.499999999999986px; top: 205.93266739736606px;\\">35</div><div class=\\"timepicker-tick\\" style=\\"left: 24.067332602633954px; top: 167.50000000000006px;\\">40</div><div class=\\"timepicker-tick\\" style=\\"left: 10px; top: 115.00000000000003px;\\">45</div><div class=\\"timepicker-tick\\" style=\\"left: 24.067332602633954px; top: 62.499999999999986px;\\">50</div><div class=\\"timepicker-tick\\" style=\\"left: 62.49999999999994px; top: 24.067332602633968px;\\">55</div></div></div><div class=\\"timepicker-footer\\"><button class=\\"btn-flat timepicker-clear waves-effect\\" style=\\"visibility: hidden;\\" type=\\"button\\" tabindex=\\"3\\">Clear</button><div class=\\"confirmation-btns\\"><button class=\\"btn-flat timepicker-close waves-effect\\" type=\\"button\\" tabindex=\\"3\\">Cancel</button><button class=\\"btn-flat timepicker-close waves-effect\\" type=\\"button\\" tabindex=\\"3\\">Ok</button></div></div></div></div></div><input name=\\"lunchtime\\" type=\\"text\\" id=\\"lunchtime\\" placeholder=\\"Choose lunchtime\\" class=\\"\\" value=\\"default value\\"><label for=\\"lunchtime\\"></label></div><button class=\\"waves-effect waves-light btn\\">Finish</button></form>"`;
